### ISSUE TITLE
createRemoteFileNode re-downloads unchanged files (the `If-None-Match` header is not being sent)

### ISSUE THREAD

**Issue by skinandbones** [gatsbyjs/gatsby#12053]
## Description

When using `createRemoteFileNode` from `gatsby-source-filesystem`, remote files get downloaded each time `yarn develop` is restarted (or each build) even though the remote files are unchanged and the cache is present.

There are multiple issues in `createRemoteFileNode`.

1. The `headers` option is not being set properly on the `got` HTTP request and, thus, the `If-None-Match` header is not getting sent.
2. When (1) is fixed, you'll find that the cached response header gets overwritten by the 304 response headers and you lose the Etag.  We should only encache response headers from a 200 response.

I'm uncertain how to wrap this up in tests but both patches are trivial and I can submit PR.

### Steps to reproduce

1. Create a new project
2. Add the following to `gatsby-node.js`

```
const { createRemoteFileNode } = require('gatsby-source-filesystem');

exports.sourceNodes = async ({ actions, store, cache, createNodeId }) => {
  const { createNode } = actions

  fileNode = await createRemoteFileNode({
    url: "https://www.nasa.gov/centers/goddard/images/content/638831main_globe_east_2048.jpg",
    cache,
    store,
    createNode,
    createNodeId
  });
}
```
3. `yarn develop`
4. Observe the file will download and you can check in cache
```
$ ls -alt .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
-rw-r--r--  1 blake  staff  536295 Feb 24 21:23 .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
$ cksum .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
1844973668 536295 .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
```

5. Wait a few minutes and restart `yarn develop`.
6. Check the cached remote files

```
$ ls -alt .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
-rw-r--r--  1 blake  staff  536295 Feb 24 21:24 .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
$ cksum .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
1844973668 536295 .cache/gatsby-source-filesystem/0d5a6ba9af7781580b3c59af9e560259/638831main_globe_east_2048.jpg
```

### Expected result

I expect the cached file to be unchanged after building (i.e., not downloaded again).

The request should send the `If-None-Match` header, the response should be a 304, and the cached file unchanged.  Subsequent restarts should continue to use the cached file.

### Actual result

The file gets re-downloaded.  You can confirm by checking the file mtime.  Secondly, I've confirmed with logging in createRemoteFileNode that the file is being re-downloaded.

### Environment

```

  System:
    OS: macOS 10.14.3
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 11.10.0 - /usr/local/bin/node
    Yarn: 1.13.0 - /usr/local/bin/yarn
    npm: 6.7.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.13 - /usr/local/bin/python
  Browsers:
    Chrome: 72.0.3626.109
    Firefox: 65.0.1
    Safari: 12.0.3
  npmPackages:
    gatsby: ^2.1.17 => 2.1.17
    gatsby-image: ^2.0.29 => 2.0.29
    gatsby-plugin-manifest: ^2.0.19 => 2.0.19
    gatsby-plugin-offline: ^2.0.24 => 2.0.24
    gatsby-plugin-react-helmet: ^3.0.6 => 3.0.6
    gatsby-plugin-sharp: ^2.0.22 => 2.0.22
    gatsby-source-filesystem: ^2.0.22 => 2.0.22
    gatsby-transformer-sharp: ^2.1.14 => 2.1.14
```

**Comment by KyleAMathews** [comment]
A PR would be great!

**Comment by skinandbones** [comment]
@KyleAMathews PR is good to go 👍

**Comment by sidharthachatterjee** [comment]
@skinandbones Your fix was just published in `gatsby-source-filesystem@2.0.23` 🙂
